gui/test/integration/cypress/support/page_files/NodeSearchResultsPage.ts

```diff
@@ -0,0 +1,30 @@
+import {Frame} from "./Frame";
+
+export class NodeSearchResultsPage extends Frame {
+  visit() {
+    cy.visit("/");
+    return this;
+  }
+
+  getVisualizationContainer() {
+    return cy.get("[data-cy=visualizationContainer]");
+  }
+
+  getNodeResultsTable() {
+    return cy.get("[data-cy=matchingNodesTable]");
+  }
+
+  getNodeResultsTableRow(index: number) {
+    return this.getNodeResultsTable().find("tbody>tr").eq(index);
+  }
+
+  /**
+   * Clicks on link for node in search results table
+   * @param index Starts at 0
+   */
+  clickNodeLinkByResultIndex(index: number) {
```

Comment:
Methods that actually take actions are good with the verb prefixes, those.

In this case I'd have a fluent interface rather than multiple methods that take the row index.

```
page.resultRow(index).nodeLink.click()
```

Prettier, or?
Of course it means creating some new types, but TypeScript doesn't make that too painful.